test/recipes/80-test_cmp_http.t

```diff
@@ -267,37 +267,30 @@ sub load_tests {
 sub start_mock_server {
     my $args = $_[0]; # optional further CLI arguments
     my $dir = bldtop_dir("");
-    local $ENV{LD_LIBRARY_PATH} = $dir;
-    local $ENV{DYLD_LIBRARY_PATH} = $dir;
+    local $ENV{LD_LIBRARY_PATH} = "$ENV{LD_LIBRARY_PATH}.:$dir";
+    local $ENV{DYLD_LIBRARY_PATH} = "$ENV{DYLD_LIBRARY_PATH}:$dir";
     my $cmd = bldtop_dir($app) . " -config server.cnf $args";
+    print "LD_LIBRARY_PATH = $ENV{LD_LIBRARY_PATH}\n";
+    print "DYLD_LIBRARY_PATH = $ENV{DYLD_LIBRARY_PATH}\n";
```

Comment:
You know, there is a OpenSSL::Test function that will give you a functioning command line that does all this already.

`cmdstr`

The proper call is this:

``` perl
    my $cmd = cmdstr(app(['openssl', '-config', 'server.cnf', $args ? $args: ()]), display => 1);
```

With that, there's no need for you to fiddle with `LD_LIBRARY_PATH` and `DYLD_LIBRARY_PATH`

Reply:
Good to learn about this nice solution!
Making use of this now.

Reply:
With this, I think you've found a bug in File::Spec...  I have a way to circumvent that problem, that will be a separate PR

Reply:
So I presume this implies we can ignore the below CI failure on **out-of-source-and-install (ubuntu-latest)**.

Reply:
Fix in #15644

Reply:
I'll be perfectly satisfied if you cherry-pick #15644 here...

Reply:
Sigh, using `cmdstr` creates two processes, e.g.,:
```
david    23315  [...] perl ../../../../util/wrap.pl ../../../../apps/openssl cmp -config server.cnf
david    23316  [...] ../../../../apps/openssl cmp -config server.cnf
```
and then
```
Killing mock server with pid=23315
```
hangs.
When I manually kill also 23316, the tests terminate successfully.

This issue also makes the CI runs hang, while unfortunately no output is shown for `make test` (which does not use the `-s` flag or the like which might explain silence).

Reply:
I just discovered that myself :-/

I tried this, to ensure that the process group gets killed, but to no avail:

diff --git a/test/recipes/80-test_cmp_http.t b/test/recipes/80-test_cmp_http.t
index e4f434678c..408f273b9d 100644
--- a/test/recipes/80-test_cmp_http.t
+++ b/test/recipes/80-test_cmp_http.t
@@ -294,5 +294,5 @@ sub start_mock_server {
 sub stop_mock_server {
     my $pid = $_[0];
     print "Killing mock server with pid=$pid\n";
-    kill('QUIT', $pid) if $pid;
+    kill('-QUIT', $pid) if $pid;
 }

Reply:
I independently digged in the same direction.
The following does not work either, because it kills also the test driver:
```
        my $pgid = `ps -o pgid= $pid | grep -o '[0-9]*'`;
        print "Killing mock server with pid=$pid, pgid=$pgid\n";
        kill -QUIT, $pgid if $pgid;
```

Reply:
Added your commit from #15644,
and a hack: kill (pid+1)

Reply:
> and a hack: kill (pid+1)

That's nice for testing the idea but unfortunately not a mergeable thing.

Reply:
> > and a hack: kill (pid+1)
> 
> That's nice for testing the idea but unfortunately not a mergeable thing.

Sure. 
Does anyone know of a simple way to kill a sub-process of a given PID, with Perl or via the shell?

Reply:
```
        my $pgid = `ps -o pgid= $pid | grep -o '[0-9]*'`;
```

!!!  Please learn to look for perl functions first!  There is one for these things, it's called [`getpgrp()`](https://perldoc.perl.org/5.10.0/functions/getpgrp)

Reply:
Good point in general, but this does not help here.
We need a way of selecting/killing only a subprocess (or all of those, here it is just one).

Reply:
I also considered using `-max_msgs`, which limits the number of messages the test server accepts before it stops, 
but this is fragile.

Reply:
https://metacpan.org/pod/Proc::Killfam ?

Reply:
Unfortunately this is not a core perl package.

Reply:
This topic is meanwhile solved, due to the below hint by @richsalz.